simulationdataschema/model_method.py

```diff
@@ -34,19 +34,1646 @@
+    A base section used to define the parameters used in the self-consistent field (SCF) simulations.
+    This section is useful to determine if a simulation `is_converged` or not (see outputs.py) and if
+    the simulation fulfills:
+
+        1. The number of iterations is not larger or equal than `n_max_iterations`.
+
+    and one of the following:
+
+        2a. The total energy change between two subsequent self-consistent iterations is below
+        `threshold_energy_change`.
+        2b. The charge density change between two subsequent self-consistent iterations is below
+        `threshold_charge_density_change`.
+    """
+
+    # TODO add examples or MEnum?
+    scf_minimization_algorithm = Quantity(
+        type=str,
+        description="""
+        Specifies the algorithm used for self consistency minimization.
+        """,
+    )
+
+    n_max_iterations = Quantity(
+        type=np.int32,
+        description="""
+        Specifies the maximum number of allowed self-consistent iterations. The simulation `is_converged`
+        if the number of iterations is not larger or equal than this quantity.
+        """,
+    )
+
+    # ? define class `Tolerance` for the different Scf tolerances types?
+    threshold_energy_change = Quantity(
+        type=np.float64,
+        unit="joule",
+        description="""
+        Specifies the threshold for the total energy change between two subsequent self-consistent iterations.
+        The simulation `is_converged` if the total energy change between two SCF cycles is below
+        this threshold.
+        """,
+    )
+
+    threshold_charge_density_change = Quantity(
+        type=np.float64,
+        description="""
+        Specifies the threshold for the average charge density change between two subsequent
+        self-consistent iterations. The simulation `is_converged` if the charge density change
+        between two SCF cycles is below this threshold.
+        """,
+    )
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        # Set the name of the section
+        self.name = self.m_def.name if self.name is None else self.name
+
+
+class BasisSet(NumericalSettings):
+    """"""
+
+    # TODO work on this base section (@ndaelman-hu)
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        # Set the name of the section
+        self.name = self.m_def.name if self.name is None else self.name
 
 
 class ModelMethod(ArchiveSection):
-    """ """
+    """
+    Model method input parameters and numerical settings used to simulate the materials properties.
+
+    # ! add more description once section is finished
+    """
 
     normalizer_level = 1
 
+    name = Quantity(
+        type=str,
+        description="""
+        Name of the model method. This is typically used to easy identification of the `ModelMethod` section.
+        Possible values: "DFT", "TB", "BSE", "GW", "DMFT".
```

Comment:
I think @ndaelman-hu replied well above.

Just to give a real case example, there was this user who wanted "kMC" covered as a label. With this, he can do it, while if we enforce MEnum, they still have to pass through us.

I will anyways add `suggestions`. I think this is possible in the ELNs, let me check.

Reply:
Ok, it does not work. So I can improve a bit the description and talk to Area D to implement this, perhaps.

Reply:
@JosePizarro3 Can elaborate on what didn't work?
If it's just the ELN suggestions, I think we can look into that separately with Area D. This can be merged.

I didn't find this piece of code (marked as outdated) anymore. Did you end up removing it?

Reply:
I tried adding:
```python
    ...
    a_eln=ELNAnnotation("StringEditQuantity", props=dict(suggestions=["DFT", "TB", "GW", "BSE", "DMFT", "kMC", "NMR"]))
    ...
```

But the develop is a bit buggy, so I couldn't see whether this was a real problem or something else going on.